cvapipe_analysis/steps/compute_features/compute_features_tools.py

```diff
@@ -0,0 +1,230 @@
+            features[f'roundness_surface_area{suffix}'] = get_surface_area(img)
+
+    else:
+        # If no foreground pixels are found
+        for img, suffix in zip([input_image,input_image_lcc],['','_lcc']):
+            features[f'shape_volume{suffix}'] = np.nan
+            features[f'position_depth{suffix}'] = np.nan
+            for uname in ['x','y','z']:
+                features[f'position_{uname}_centroid{suffix}'] = np.nan
+            features[f'roundness_surface_area{suffix}'] = np.nan
+        
+    if not compute_shcoeffs:
+        features = cast_features(features)
+        return features
+    
+    # Spherical harmonics expansion
+    angle = np.nan
+    if input_reference_image is not None:
+
+        # Get alignment angle based on the reference image. Variance
+        # paper uses make_unique = False
+        input_ref_image_aligned, angle = shtools.align_image_2d(
+            image = input_reference_image,
+            make_unique = False
+        )
+                
+        # Rotate input image according the reference alignment angle
+        input_image_lcc_aligned = shtools.apply_image_alignment_2d(
+            image = input_image_lcc,
+            angle = angle
+        ).squeeze()
+                
+    else:
+        
+        input_image_lcc_aligned = input_image_lcc
+        
+    (coeffs, _), (_, _, _, transform) = shparam.get_shcoeffs(
+        image = input_image_lcc_aligned,
+        lmax = 16,
+        sigma = 2,
+        alignment_2d = False
+    )
+    
+    if transform is not None:
+        transform = {
+            'shcoeffs_transform_xc_lcc': transform[0],
+            'shcoeffs_transform_yc_lcc': transform[1],
+            'shcoeffs_transform_zc_lcc': transform[2],
+            'shcoeffs_transform_angle_lcc': angle,
+        }
+    else:
+        transform = {
+            'shcoeffs_transform_xc_lcc': np.nan,
+            'shcoeffs_transform_yc_lcc': np.nan,
+            'shcoeffs_transform_zc_lcc': np.nan,
+            'shcoeffs_transform_angle_lcc': np.nan,
+        }
+
+    # Add suffix to identify coeffs have been calculated on the
+    # largest connected component
+    coeffs = dict(
+        (f'{key}_lcc',value) for (key,value) in coeffs.items()
+    )
+        
+    features.update(coeffs)
+    features.update(transform)
+
+    for key, value in features.items():
+        features = cast_features(features)
+    
+    return features
+
+def get_segmentations(folder, path_to_seg, channels):
+
+    """
+    Find the segmentations that should be used for features
+    calculation.
+
+    Parameters
+    --------------------
+    path_to_seg: str
+        Path to the 4D binary image.
+    channels: list of str
+        Name of channels of the 4D binary image.
+
+    Returns
+    -------
+    result: seg_nuc, seg_mem, seg_str
+        3D binary images of nucleus, cell and structure.
+    """
+
+    ch_dna = channels.index('dna_segmentation')
```

Comment:
Will do.

Reply:
maybe a good idea is to specify all channel names in a class in a separate file so we are explicit about it

Reply:
Ok. I will create a YAML config file to be shared across all steps. For now it will be a new issue: https://github.com/AllenCell/cvapipe_analysis/issues/5#issue-807619785.